controllers/announcements.go

```diff
@@ -0,0 +1,98 @@
+package controllers
+
+import (
+    "bytes"
+    "strconv"
+    "net/url"
+    "net/http"
+    "io/ioutil"
+    "encoding/json"
+    "gopkg.in/macaron.v1"
+)
+
+type Topic struct {
+    ID int `json:"id"`
+    Title string `json:"title"`
+    Slug string `json:"slug"`
+}
+
+type TopicList struct {
+    Topics []Topic `json:"topics"`
+}
+
+type Category struct {
+    TopicList TopicList `json:"topic_list"`
+}
+
+type Post struct {
+    Cooked string `json:"cooked"`
+}
+
+type PostStream struct {
+    Posts []Post `json:"posts"`
+}
+
+type TopicResponse struct {
+    PostStream PostStream `json:"post_stream"`
+}
+
+type AnnouncementView struct {
+    First Announcement `json:"first"`
+    Second Announcement `json:"second"`
+}
+
+type Announcement struct {
+    Title string `json:"title"`
+    Content string `json:"content"`
+    Link string `json:"link"`
+}
+
+func GetAnnouncements(ctx *macaron.Context) {
+    data := url.Values{}
+    client := &http.Client{}
```

Comment:
Why do you need a custom client, if you don't set anything (from what I can tell), the `http` package has a default client you can use, together with a useful global method `http.Get(url)`